pose_prediction/plugin.cpp

```diff
@@ -10,31 +12,62 @@ class pose_prediction_impl : public pose_prediction {
     pose_prediction_impl(const phonebook* const pb)
 		: sb{pb->lookup_impl<switchboard>()}
 		, _m_slow_pose{sb->subscribe_latest<pose_type>("slow_pose")}
+        , _m_imu_biases{sb->subscribe_latest<imu_biases_type>("imu_biases")}
         , _m_true_pose{sb->subscribe_latest<pose_type>("true_pose")}
     { }
 
-    virtual pose_type get_fast_pose() const override {
-		const pose_type* pose_ptr = _m_slow_pose->get_latest_ro();
-		return correct_pose(
-			pose_ptr ? *pose_ptr : pose_type{}
-		);
+    // No paramter pose predict will just get the current slow pose based on the next vsync
+    virtual fast_pose_type get_fast_pose() override {
+		return get_fast_pose(std::chrono::high_resolution_clock::now());
     }
 
-    virtual pose_type get_true_pose() const override {
+    virtual pose_type get_true_pose() override {
 		const pose_type* pose_ptr = _m_true_pose->get_latest_ro();
 		return correct_pose(
 			pose_ptr ? *pose_ptr : pose_type{}
 		);
     }
 
+    // future_time: An absolute timepoint in the future
+    virtual fast_pose_type get_fast_pose(time_type future_timestamp) override {
+
+        if (!_m_imu_biases->get_latest_ro()) {
+            const pose_type* pose_ptr = _m_slow_pose->get_latest_ro();
+            
+            return fast_pose_type{
+                .pose = correct_pose(pose_ptr ? *pose_ptr : pose_type{})
```

Comment:
Why does this not set the computed time and target time? The caller has no way of knowing what pose prediction did, and will use the times regardless.